language/move-vm/types/src/values/values_impl.rs

```diff
@@ -1260,36 +1260,52 @@ impl IntegerValue {
     }
 }
 
-impl From<IntegerValue> for u8 {
-    fn from(value: IntegerValue) -> u8 {
+macro_rules! narrowing_cast {
+    ($wide_ty: ident, $narrow_ty: ident, $x: expr) => {{
+        let x = $x;
+        if x > (std::$narrow_ty::MAX as $wide_ty) {
+            let msg = format!(
+                "Cannot cast {}({}) to {}",
+                stringify!($wide_ty),
+                x,
+                stringify!($narrow_ty)
+            );
+            Err(VMStatus::new(StatusCode::ARITHMETIC_ERROR).with_message(msg))
+        } else {
+            Ok(x as $narrow_ty)
+        }
+    }};
+}
```

Comment:
Here I could have used some 3rd party crates, but decided not to.
- `num_traits` allows me to rewrite this macro as a generic function, but also introduces many other traits we don't currently use
- `cast` implements checked conversion between any integer and float types Rust has, most of which we don't use

Therefore I'd be reluctant to introduce additional dependencies to the TCB unless necessary. (If we add a dependency, ideally, we will need to audit it very carefully.)

Reply:
- Agreed with not using third-party crates
- I think it would be more readable to write this code a few times than it would be to use a macro

Reply:
Got the macro removed.